### DISCUSSION TITLE
Client Disconnect - Webhook

### DISCUSSION THREAD

**Discussion by abhigyan231195** [deepgram/community#827] [General help]
### Which Deepgram product feature are you using?

Deepgram API - STT Pre-recorded

### Details

When I submit audio files larger than 3 hours then quite frequently Deepgram calls the callback webhook URL but I get Client Disconnect error as soon as I try to get the json out of payload.
I am using FastAPI container for hosting webhook endpoints. Is anyone aware of this?

<img width="669" alt="image" src="https://github.com/deepgram/community/assets/122435030/f4c3cbec-ab43-4071-a19c-0c74d0a8f2df">

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

https://api.deepgram.com/v1/listen?language=en-US&punctuate=true&profanity_filter=true&diarize=false&smart_format=false

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

62111ce5-9f63-4c52-a116-1bfa5b564d3b

### If possible, please attach your code or paste it into the text box.

```
@webhook_router_v1.post("/deepgram")
async def deepgram_webhook(request: Request, background_tasks: BackgroundTasks):
    """
        Deepgram webhook handler
    """
    try:
        body = await request.json()
    except Exception as e:
        logger.error(f"Error parsing request body: {e}")
        raise HTTPException(status_code=400, detail="Error parsing request body")

    deepgram = DeepgramHandler(body)
    background_tasks.add_task(deepgram.process)
    return {"status": "success"}
```

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by SandraRodgers** [comment]
So this only happens some of the time? Is it possible that the server where FastAPI is running might have a payload limit, such as these servers do? https://developers.deepgram.com/docs/payload-too-large

**Comment by abhigyan231195** [comment]
@SandraRodgers: This happens few times not always and no we have not set any payload limit in FastAPI nor in the Gunicorn ASGI server which runs our FastAPI container.
